pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -373,3 +425,71 @@ int64_t     *numpy_to_ndarray_shape(int64_t *np_shape, int nd)
     return nd_shape;
 
 }
+
+void array_copy_data(t_ndarray dest, t_ndarray src, int64_t offset)

```

Comment:
maybe you could use memcpy in the case of order C

Reply:
Yes, that will be faster if both the arrays are of the same order.
Actually it would also work for `order_f` arrays.

Reply:
Okey, for `order_f` it will only work if the array being copied is not a list of arrays.
But it is a good optimization, I shall do it.

Reply:
Maybe, for performance's sake, we should have a specific function for copying that uses memcopy when possible (although the overhead of the copying does not hurt the instant memory accessing), and another for slicing.

Reply:
@soukainahikma when  slicing, can we have it so that the new `t_ndarray` points to the same place in memory of the sliced array, but has different `strides, shape, nd....`, this should make it so that we don't have to copy the array at all

Reply:
Slicing creates views, but fancy slicing creates copies
[source](https://scipy-cookbook.readthedocs.io/items/ViewsVsCopies.html)